clang/include/clang/CIR/Dialect/IR/CIROps.td

```diff
@@ -4218,32 +4218,44 @@ def MemChrOp : CIR_Op<"libc.memchr"> {
 }
 
 //===----------------------------------------------------------------------===//
-// ReturnAddrOp
+// FuncAddrBuiltinOp
 //===----------------------------------------------------------------------===//
 
-def ReturnAddrOp : CIR_Op<"return_address"> {
-  let arguments = (ins UInt32:$level);
-  let summary = "The return address of the current function, or of one of its callers";
+def ReturnAddr : I32EnumAttrCase<"return_address", 1>;
+def FrameAddr : I32EnumAttrCase<"frame_address", 2>;
+def FuncAddrKind
+    : I32EnumAttr<"FuncAddrKind",
+                  "address kind returned from builtin for function",
+                  [ReturnAddr, FrameAddr]> {
+  let cppNamespace = "::cir";
+}
+
+def FuncAddrBuiltinOp : CIR_Op<"func_addr_builtin"> {
+  let arguments = (ins Arg<FuncAddrKind, "address kind">:$kind, UInt32:$level);
+  let summary = "Builtin that return function return or frame address";
   let results = (outs Res<VoidPtr, "">:$result);
 
   let description = [{
-    Represents call to builtin function ` __builtin_return_address` in CIR.
-    This builtin function returns the return address of the current function,
-    or of one of its callers.
+    Represents call to builtin function such as `__builtin_return_address` 
+    or `__builtin_frame_address` in CIR.
+    This builtin function returns the return or frame address of 
+    the current function, or of one of its callers.
     The `level` argument is number of frames to scan up the call stack.
-    For instance, value of 0 yields the return address of the current function,
-    value of 1 yields the return address of the caller of the current function,
-    and so forth.
+    For instance, 
```

Comment:
The line wrapping here is a bit strange?

Reply:
I was thinking to make contrast of  `value of 0` and `value of 1` lines more contrastingly visible. But open for change it to more normal line wrapping it feels strange.